evalml/data_checks/data_check_action.py

```diff
@@ -0,0 +1,24 @@
+class DataCheckAction:
+    """Base class for all DataCheckActions."""
+
+    def __init__(self, action_code, details=None):
+        """
+        A recommended action returned by a DataCheck.
+
+        Arguments:
+            action_code (DataCheckActionCode): Action code associated with the action.
+            details (dict, optional): Additional useful information associated with the action
+        """
+        self.action_code = action_code
+        self.details = details
+
+    def __eq__(self, other):
+        """Checks for equality. Two DataCheckAction objs are considered equivalent if all of their attributes are equivalent."""
+        return (self.action_code == other.action_code and
+                self.details == other.details)
+
+    def to_dict(self):
+        action_dict = {"code": self.action_code.name}
+        if self.details is not None:
+            action_dict.update({"details": self.details})
```

Comment:
I think it's better if `details` is always present and just set to an empty dict if there are no details.

Just so that users don't have to check if there are details before trying to access them.